### PR TITLE
Fix UI Header Response to Browser and Crossfeed Logo Routing

### LINKED ISSUE #16: UI: Adjust Header to Response to Browser
# 💡 Summary #

The current header in our React Material UI application needs to be responsive to different browser sizes. We want to ensure that it looks good on both desktop and mobile devices. This issue aims to adjust the header component to achieve responsiveness.

## Motivation and context ##

Responsive Design: The header should adapt its layout based on the screen size. On smaller screens (e.g., mobile devices), the header should collapse into a hamburger menu or a more compact version. On larger screens (e.g., desktop), the header should display the full navigation menu.

Current Layout
Screenshot 2024-03-06 at 7 13 25 PM

## Implementation notes ##

- Use Material UI’s breakpoints (e.g., xs, sm, md, lg, xl) to define different header layouts.
- Consider adjusting font sizes, spacing, and alignment for each breakpoint.

## Acceptance criteria ##

- [ ] Commit to Integration Branch
- [ ] Demonstrate a Response Design
- [ ] Include all MUI breakpoints
- [ ] Include Screenshots Of Updated Views
- [ ] Commit to Staging
- [ ] Create Functional Tests

### LINKED ISSUE #17: UI: Set Crossfeed Logo Link to Home
# 💡 Summary #

The current behavior of our application’s menu logo link directs users to various pages or no page at all. We want to update it so that clicking the logo always navigates users to the home page (Dashboard). 

## Motivation and context ##

This issue aims to modify the logo link behavior accordingly. This effort can be consolidated with https://github.com/cisagov/XFD/issues/16

Logo
Screenshot 2024-03-06 at 7 27 06 PM

## Implementation notes ##

- Verify that the logo link works with responsive UI
- Apply the DRY principle

## Acceptance criteria ##

How do we know when this work is done?

- [ ]  Commit to Integration Branch
- [ ] Demonstrate a Response Design
- [ ] Include Screenshots Of Updated Views
- [ ] Commit to Staging
- [ ] Create Functional Tests

### PR DESCRIPTION
**Pull request by chrtorres**
#  #

## 🗣 Description ##

- Fixed an issue preventing responsiveness from the UI header.
- Adjusted NavLink styling.
- Corrected Crossfeed logo routing incorrectly.

## 💭 Motivation and context ##
Closes issues 16 & 17.

## 🧪 Testing ##

Sign in as different user types and see if the header is correctly displayed and responsive.

## 📷 Screenshots ##

**Full screen design**

image

image

**Small screen design**

image

image

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: frontend/src/components/Header.tsx
Comment (Matthew-Grayson) [inline frontend/src/components/Header.tsx:1]: There are lint issues flagged by Github Actions.

Comment (chrtorres) [reply frontend/src/components/Header.tsx:1]: Lint issues have been addressed and committed.

**Review by cduhn17 — APPROVED** [review]
LGTM

**Review by nickviola — APPROVED** [review]
LGTM

**Comment by nickviola** [comment]
Tested in both Safari and Chrome. 

Chrome:
Screenshot 2024-03-27 at 1 38 27 PM


Safari:
Screenshot 2024-03-27 at 1 38 39 PM


All tested and works as expected.

**Pull request merged**